custom_components/unfoldedcircle/config_flow.py

```diff
@@ -796,27 +796,52 @@ async def async_step_select_entities(
                 menu_options=["error", "finish"],
             )
         _LOGGER.debug(
-            "Found configuration subscription for remote : %s",
-            configure_entities_subscription,
+            "Found configuration subscription for remote %s (subscription_id %s) : entities %s",
+            configure_entities_subscription.client_id,
+            configure_entities_subscription.subscription_id,
+            configure_entities_subscription.entity_ids
         )
         subscribed_entities: list[str] = []
         if subscribed_entities_subscription:
             _LOGGER.debug(
-                "Found subscribed entities for remote : %s",
-                subscribed_entities_subscription,
+                "Found subscribed entities for remote %s (subscription_id %s) : %s",
+                subscribed_entities_subscription.client_id,
+                subscribed_entities_subscription.subscription_id,
+                subscribed_entities_subscription.entity_ids
             )
             subscribed_entities = subscribed_entities_subscription.entity_ids
 
+        # Initialize the available entities from : subscribed entities + available entities stored in config entry
+        # (if any)
+        available_entities = subscribed_entities.copy()
+
+        # Only in option flow : retrieve configured available entities stored in the integration
+        # and add them to the list if not present
+        if (isinstance(config_flow, UnfoldedCircleRemoteOptionsFlowHandler) and config_flow.options
+                and config_flow.options.get("available_entities", None)):
+            entities = config_flow.options["available_entities"]
+            for entity_id in entities:
+                if entity_id not in available_entities:
+                    available_entities.append(entity_id)
+
+        # Selector for entities to add (all except those already in the available list
         config: EntitySelectorConfig = {
-            "exclude_entities": subscribed_entities,
+            "exclude_entities": available_entities,
             "filter": [{"domain": filtered_domains}],
             "multiple": True,
         }
-
         data_schema: dict[any, any] = {"add_entities": EntitySelector(config)}
-        if len(subscribed_entities) > 0:
+
+        # Selector for entities to be removed from available list :
+        # all in available list except those already subscribed which should be kept in the list
+        removable_list = available_entities.copy()
+        for entity_id in subscribed_entities:
```

Comment:
I think subscribed_entities is ok, but just double check you didn't mean available_entities

Reply:
no this is correct : in the removable list (which contain entities that can be removed), we can have all available entities EXCEPT the ones that are subscribed. So I "remove" from the "removable_list" the subscribed entities